### PR TITLE
feat(maxTimeMS): adding maxTimeMS option to createIndexes and dropIndexes

### PR DESCRIPTION
**Pull request by daprahamian**

### CODE REVIEW

File: lib/collection.js [hunk 1]
```diff
@@ -1695,12 +1698,15 @@ define.classMethod('dropIndex', { callback: true, promise: true });
...
+ * @param {number} [options.maxTimeMS=null] Number of miliseconds to wait before aborting the query.
  * @param {Collection~resultCallback} [callback] The command result callback
  * @return {Promise} returns Promise if no callback passed
  */
 Collection.prototype.dropIndexes = function(options, callback) {
   if (typeof options === 'function') (callback = options), (options = {});
-  options = options || {};
+  options = options ? shallowClone(options) : {};
+
+  if (typeof maxTimeMS !== 'number') delete options.maxTimeMS;
```
Comment (mbroadst) [inline lib/collection.js]: same issue here as above

File: lib/collection.js [hunk 2]
```diff
@@ -1608,7 +1608,9 @@ define.classMethod('createIndex', { callback: true, promise: true });
  */
 Collection.prototype.createIndexes = function(indexSpecs, options, callback) {
   if (typeof options === 'function') (callback = options), (options = {});
-  options = options || {};
+
+  options = options ? shallowClone(options) : {};
+  if (typeof maxTimeMS !== 'number') delete options.maxTimeMS;
```
[the remark this answers is not in the document]
Comment (daprahamian) [reply lib/collection.js]: ah, yeah, that's left over from a previous version. I'll get up a fix ASAP

File: lib/collection.js [hunk 3]
```diff
@@ -1659,6 +1661,7 @@ define.classMethod('createIndexes', { callback: true, promise: true });
  * @param {number} [options.wtimeout=null] The write concern timeout.
  * @param {boolean} [options.j=false] Specify a journal write concern.
  * @param {ClientSession} [options.session] optional session to use for this operation
+ * @param {number} [options.maxTimeMS=null] Number of miliseconds to wait before aborting the query.
```
[the remark this answers is not in the document]
Comment (daprahamian) [reply lib/collection.js]: I removed the `=null`, but I am wondering if there should be a default value for this? Do we have a default timeout?

**Comment by daprahamian** [comment]
All changes incorporated.

Side note, I am concerned that the ESLint did not pick up the `maxTimeMS` error.

**Pull request merged**